core/src/main/java/org/apache/brooklyn/core/mgmt/ha/OsgiArchiveInstaller.java

```diff
@@ -0,0 +1,429 @@
+        } finally {
+            zipIn = null;
+        }
+    }
+
+    private void discoverManifestFromCatalogBom(boolean isCatalogBomRequired) {
+        discoveredManifest = new BundleMaker(mgmt()).getManifest(zipFile);
+        ZipFile zf = null;
+        try {
+            try {
+                zf = new ZipFile(zipFile);
+            } catch (IOException e) {
+                throw new IllegalArgumentException("Invalid ZIP/JAR archive: "+e);
+            }
+            ZipEntry bom = zf.getEntry("catalog.bom");
+            if (bom==null) {
+                bom = zf.getEntry("/catalog.bom");
+            }
+            if (bom==null) {
+                if (isCatalogBomRequired) {
+                    throw new IllegalArgumentException("Archive must contain a catalog.bom file in the root");
+                } else {
+                    return;
+                }
+            }
+            String bomS;
+            try {
+                bomS = Streams.readFullyString(zf.getInputStream(bom));
+            } catch (IOException e) {
+                throw new IllegalArgumentException("Error reading catalog.bom from ZIP/JAR archive: "+e);
+            }
+            discoveredBomVersionedName = BasicBrooklynCatalog.getVersionedName( BasicBrooklynCatalog.getCatalogMetadata(bomS), false );
+        } finally {
+            Streams.closeQuietly(zf);
+        }
+    }
+    
+    private void updateManifestFromAllSourceInformation() {
+        if (discoveredBomVersionedName!=null) {
+            matchSetOrFail("catalog.bom in archive", discoveredBomVersionedName.getSymbolicName(), discoveredBomVersionedName.getVersion().toString());
+        }
+        
+        boolean manifestNeedsUpdating = false;
+        if (discoveredManifest==null) {
+            discoveredManifest = new Manifest();
+            manifestNeedsUpdating = true;
+        }
+        if (!matchSetOrFail("MANIFEST.MF in archive", discoveredManifest.getMainAttributes().getValue(Constants.BUNDLE_SYMBOLICNAME),
+                discoveredManifest.getMainAttributes().getValue(Constants.BUNDLE_VERSION) )) {
+            manifestNeedsUpdating = true;                
+            discoveredManifest.getMainAttributes().putValue(Constants.BUNDLE_SYMBOLICNAME, inferredMetadata.getSymbolicName());
+            discoveredManifest.getMainAttributes().putValue(Constants.BUNDLE_VERSION, inferredMetadata.getVersion());
+        }
+        if (Strings.isBlank(inferredMetadata.getSymbolicName())) {
+            throw new IllegalArgumentException("Missing bundle symbolic name in BOM or MANIFEST");
+        }
+        if (Strings.isBlank(inferredMetadata.getVersion())) {
+            throw new IllegalArgumentException("Missing bundle version in BOM or MANIFEST");
+        }
+        if (discoveredManifest.getMainAttributes().getValue(Attributes.Name.MANIFEST_VERSION)==null) {
+            discoveredManifest.getMainAttributes().putValue(Attributes.Name.MANIFEST_VERSION.toString(), "1.0");
```

Comment:
I think this is always "1.0" (I think it's about the version of the manifest format?), rather than being the version of the code within the jar.